Extension/src/LanguageServer/extension.ts

```diff
@@ -293,9 +294,43 @@ function realActivation(): void {
     updateLanguageConfigurations();
 
     reportMacCrashes();
-
+    
     const settings: CppSettings = new CppSettings(clients.ActiveClient.RootUri);
-    if (settings.updateChannel === 'Insiders') {
+    let suggestInsiders: PersistentState<boolean> = new PersistentState<boolean>("CPP.suggestInsiders", true);
+    
+    if (suggestInsiders.Value && settings.updateChannel === 'Default') {
+        getTargetBuildInfo("Insiders").then(buildInfo => {
+            if (!buildInfo) {
+                return;
+            }
+            const message: string = `The insider version (${buildInfo.name}) is available, switch to the insider channel?`;
+            const yes: string = "Yes";
+            const askLater: string = "Ask Me Later";
+            const dontShowAgain: string = "Don't Show Again";
+            vscode.window.showInformationMessage(message, yes, askLater, dontShowAgain).then(selection => {
+                switch (selection) {
+                    case yes:
+                        // Avoid double calling applyUpdate
+                        shouldCheckAndApplyUpdate = false;
+                        vscode.workspace.getConfiguration("C_Cpp").update("updateChannel", "Insiders", true);
+                        return applyUpdate(buildInfo, 'Insiders');
+                    case dontShowAgain:
+                        suggestInsiders.Value = false;
+                        break;
+                    case askLater:
+                        break;
+                    default:
+                        break;
+                }
+            });
+        }).catch(error => {
+            // Handle .then following getTargetBuildInfo rejection
+            if (error.message.indexOf('/') !== -1 || error.message.indexOf('\\') !== -1) {
+                error.message = "Potential PII hidden";
+            }
+            telemetry.logLanguageServerEvent('installVsix', { 'error': error.message, 'success': 'false' });
```

Comment:
We need to think about a different telemetry event here or simply removing this.  This telemetry event name is used for the path that actually installs the VSIX.  I'll discuss with @sean-mcmanus.

Reply:
What's wrong with this telemetry? Isn't it install related?

Reply:
Yes, but this is not for people on the Insiders channel. This is for **everyone** (at least once).  I don't want to merge telemetry from `getTargetBuildInfo` errors for people on the Default channel with people on the Insiders channel.

I think there's two options here:
1. Insert a `.catch` between `getTargetBuildInfo` and `.then` so we can filter out those errors (maybe log a different event there)
2. Move this catch to `applyUpdate(...).catch`

Reply:
What should we do with the exception thrown by `getTargetBuildInfo`, ignore it directly?

Reply:
Can you send the same telemetry but with EventName "suggestInsiders" instead of "installvsix"?